src/OpenIddict.Core/OpenIddictProvider.Authentication.cs

```diff
@@ -16,17 +15,21 @@
 using Microsoft.AspNetCore.Authentication;
 using Microsoft.AspNetCore.Http.Authentication;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
 
-namespace OpenIddict {
+namespace OpenIddict
+{
     public partial class OpenIddictProvider<TUser, TApplication> : OpenIdConnectServerProvider where TUser : class where TApplication : class {
         public override async Task ValidateAuthorizationRequest([NotNull] ValidateAuthorizationRequestContext context) {
             var services = context.HttpContext.RequestServices.GetRequiredService<OpenIddictServices<TUser, TApplication>>();
+            var logger = context.HttpContext.RequestServices.GetRequiredService<ILogger<OpenIddictProvider<TUser, TApplication>>>();
 
             // Note: redirect_uri is not required for pure OAuth2 requests
             // but this provider uses a stricter policy making it mandatory,
             // as required by the OpenID Connect core specification.
             // See http://openid.net/specs/openid-connect-core-1_0.html#AuthRequest.
             if (string.IsNullOrEmpty(context.RedirectUri)) {
+                logger.LogWarning("The required redirect_uri parameter was missing.");
```

Comment:
FYI, ASOS beta5 now logs the error code and the description you specify when calling `context.Reject(...)` so it may be redundant. Consider removing the log messages that are equivalent to the error description.